rust/tools/authorization-logic/src/souffle/souffle_emitter.rs

```diff
@@ -15,116 +15,81 @@
  */
 
 use crate::{ast::*, souffle::datalog_ir::*};
-use std::collections::HashSet;
 
-pub struct SouffleEmitter {
-    // This is a set of predicates that need to be declared. This set is
-    // populated by side effect while traversing the program body.
-    decls: HashSet<AstPredicate>,
```

Comment:
Declarations used to be generated by keeping around a set of all the predicates that were encountered and then translating every parameter into a symbol.  This is removed now that there are type declarations.

Reply:
Thanks for the heads up. In github, you can hide whitespace during review if needed.